src/S3/Parser/GetBucketLocationResultMutator.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Aws\S3\Parser;
+
+use Aws\CommandInterface;
+use Aws\ResultInterface;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A custom mutator for a GetBucketLocation request, which
+ * extract the bucket location value and injects it into the
+ * result as the `LocationConstraint` field.
+ *
+ * @internal
+ */
+final class GetBucketLocationResultMutator implements S3ResultMutator
+{
+    /**
+     * @inheritDoc
+     */
+    public function __invoke(
+        ResultInterface $result,
+        CommandInterface $command,
+        ResponseInterface $response
+    ): ResultInterface
+    {
+        if ($command->getName() !== 'GetBucketLocation') {
+            return $result;
+        }
+
+        $location = 'us-east-1';
+        if (preg_match('/>(.+?)<\/LocationConstraint>/', $response->getBody(), $matches)) {
```

Comment:
This also exceeds 80 characters.  You could probably get around that by moving the regex to a const